chroma_core/models/client_mount.py

```diff
@@ -48,12 +48,26 @@ def get_deps(self, state=None):
             deps.append(DependOn(self.host.lnet_configuration, "lnet_up", fix_state="unmounted"))
 
         if state != "removed":
-            fs = ObjectCache.get_one(ManagedFilesystem, lambda mf: mf.name == self.filesystem)
-
-            # Depend on the fs being available.
-            deps.append(DependOn(fs, "available", fix_state="unmounted"))
+            try:
+                fs = ObjectCache.get_one(ManagedFilesystem, lambda mf: mf.name == self.filesystem)
+
+                # Depend on the fs being available.
+                deps.append(DependOn(fs, "available", fix_state="unmounted"))
+
+                # If the filesystem is removed, the
+                # mount should follow.
+                deps.append(
+                    DependOn(
+                        fs,
+                        "available",
+                        acceptable_states=list(set(fs.states) - set(["removed", "forgotten"])),
+                        fix_state="removed",
+                    )
+                )
+            except ManagedFilesystem.DoesNotExist:
+                pass
 
-            # But if either the host or the filesystem are removed, the
+            # If the host is is removed, the
```

Comment:
```suggestion
            # If the host is removed, the
```

Reply:
Oops, I'll fix before landing or if I have to push again.